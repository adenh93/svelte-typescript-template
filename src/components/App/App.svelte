<script lang="ts">
  import type { HelloMessage } from '../../types'
  export let helloMessage: HelloMessage
</script>

<main>
  <h1>Hello {helloMessage.name}!</h1>
  <p>This template was last updated on {helloMessage.date}</p>
</main>
