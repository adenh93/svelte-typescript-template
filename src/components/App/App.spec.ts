import { cleanup, render } from '@testing-library/svelte'
import type { HelloMessage } from '../../types'
import App from '.'

describe('App Component', () => {
  let helloMessage: HelloMessage = {
    name: 'Test',
    date: '10/10/10',
  }

  afterEach(cleanup)

  test('should render', () => {
    const { getByText } = render(App, { helloMessage })

    expect(getByText(`Hello ${helloMessage.name}!`))
  })

  test('should match snapshot', () => {
    const { container } = render(App, { helloMessage })

    expect(container).toMatchSnapshot()
  })
})
