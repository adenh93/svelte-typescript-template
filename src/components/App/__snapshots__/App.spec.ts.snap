// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<body>
  <div>
    <main>
      <h1>
        Hello 
        Test
        !
      </h1>
       
      <p>
        This template was last updated on 
        10/10/10
      </p>
    </main>
  </div>
</body>
`;
